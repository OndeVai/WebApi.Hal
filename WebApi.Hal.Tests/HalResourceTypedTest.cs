#region

using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using ApprovalTests;
using ApprovalTests.Reporters;
using WebApi.Hal.Tests.Representations;
using Xunit;

#endregion

namespace WebApi.Hal.Tests
{
    public class HalResourceTypedTest
    {
        readonly OrganisationRepresentationTyped resource;

        public HalResourceTypedTest()
        {
            var model = new Organisation(1, "Org Name");
            var emps = new List<Employee>
            {
                new Employee(1, "emp1", 1),
                new Employee(11, "emp2", 2),
                new Employee(12, "emp1", 4),
            };

            var employeeRepresentations = emps.Select(e => new EmployeeRepresentation(e)).ToList();
            model.Employees = new EmployeeRepresentationList(employeeRepresentations);
            resource = new OrganisationRepresentationTyped(model);
        }

        [Fact]
        [UseReporter(typeof (DiffReporter))]
        public void organisation_get_json_test()
        {
            // arrange
            var mediaFormatter = new JsonHalMediaTypeFormatter {Indent = true};
            var content = new StringContent(string.Empty);
            var type = resource.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, resource, stream, content, null);
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }

        [Fact]
        [UseReporter(typeof (DiffReporter))]
        public void organisation_get_xml_test()
        {
            // arrange
            var mediaFormatter = new XmlHalMediaTypeFormatter();
            var content = new StringContent(string.Empty);
            var type = resource.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, resource, stream, content, null);
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }
    }
}