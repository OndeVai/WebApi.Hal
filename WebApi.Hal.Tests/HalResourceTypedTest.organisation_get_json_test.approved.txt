{
  "_links": {
    "self": {
      "href": "/api/employees"
    }
  },
  "_embedded": {
    "employees": [
      {
        "id": 1,
        "title": "emp1",
        "rank": 1,
        "_links": {
          "self": {
            "href": "/api/employees/1",
            "title": "see employee emp1"
          },
          "related": {
            "href": "sometoher",
            "title": "see related"
          }
        }
      },
      {
        "id": 11,
        "title": "emp2",
        "rank": 2,
        "_links": {
          "self": {
            "href": "/api/employees/11",
            "title": "see employee emp2"
          },
          "related": {
            "href": "sometoher",
            "title": "see related"
          }
        }
      },
      {
        "id": 12,
        "title": "emp1",
        "rank": 4,
        "_links": {
          "self": {
            "href": "/api/employees/12",
            "title": "see employee emp1"
          },
          "related": {
            "href": "sometoher",
            "title": "see related"
          }
        }
      }
    ]
  }
}